libs/src/services/contracts/ContractClient.js

```diff
@@ -112,7 +114,29 @@ class ContractClient {
     if (!(methodName in this._contract.methods)) {
       throw new Error(`Contract method ${methodName} not found in ${Object.keys(this._contract.methods)}`)
     }
-    return this._contract.methods[methodName](...args)
+    const method = await this._contract.methods[methodName](...args)
```

Comment:
just so i understand, this is a different layer of retries than the ones in the web3Managers. This is intended to retry the actual call function to the blockchain but not send any transactions. Essentially this is read side retries for POA and ETH. and the respective web3Managers are writes.

Reply:
yep! exactly